### PR TITLE
Create foundation for Hud Implementation.

### PR DESCRIPTION
**Pull request by Bailey-Ardrey**
I have shown this code to you all in meetings a few times. Basically, I have a Hud class for each section of the hud menu. One for the dungeon, mini-map, in-game stuff, and inventory. These are initialized as IHud objects within the Game class and are updated and drawn there.

Each Hud implementation will be different, but each will first draw the hud itself, then black squares block out any of the placeholder squares. Then, icons can be added to the hud as needed by adding an Entity to the list of Icons within each hud. These Icons are iterated through and drawn within Game. Some huds might need helper classes (like for in-game, I have a Number Manager), to extract some functionality out.

Let me know if you have any questions or suggestions!

### CODE REVIEW

**Comment by bjhobson12** [comment]
I'm not sure what to do about the conflicts, but this code works for me. The hearts are still not quite right (which I know you know). You should merge this after conflicts resolve.

**Comment by Bailey-Ardrey** [comment]
The last two commits I just added are pretty fundamental for the Hud implementation. It sets the precedent that implementation details for the Hud should reside in dedicated classes for those actions... this will decrease the amount of cohesion in the hud classes!

Firstly, I fixed the Heart implementation and put these details in a HudHeartManager class that the InGameHud has an instance of. The InGameHud then just passes the HudHeartManager Link's current health and max health and the manager does the rest.

Secondly, I created a HudInitializer class that all the Huds should be able to use to initialize their squares. Each hud type will need to have its own method tho, because initialization is specific to each one.

File: Sprintfinity3902/PauseMenu.cs
Comment (kylegessel) [inline Sprintfinity3902/PauseMenu.cs:17]: Is the map no longer needed for the pause menu? whats the change here?

**Comment by Bailey-Ardrey** [comment]
Okay, one last commit. I just moved the MiniMap initialization to the HudInitializer class to keep things consistent. This way everyone can follow that format.

File: Sprintfinity3902/Sprintfinity3902.csproj
Comment (kylegessel) [inline Sprintfinity3902/Sprintfinity3902.csproj:17]: This is strange to me, I just have no idea why this would have changed.

**Comment by bjhobson12** [comment]
Bailey I ran this and it looks good. You should remember that we aren't supposed to have magic numbers... It's not helpful to have to rework old code for it so just remember it in classes you create. If you want to redo that you can or just wait for someone to merge this. I'm holding off only because I'm not sure what best action is. 

I'm going to pull from your branch regardless.

**Comment by Bailey-Ardrey** [comment]
It might be helpful to have someone hop on a call with me to discuss magic numbers. I might not fully understand what constitutes a magic number, and it would be helpful to walk through my own code with someone else to learn more so I can correct this moving forward. Would someone be willing to take a look with me?

**Review by kylegessel — APPROVED** [review]
109 files!!! Most were directive cleanup or new sprites. I love how the HUDs have been setup! Very clean.

File: Sprintfinity3902/Sprintfinity3902.csproj
Comment (Bailey-Ardrey) [reply Sprintfinity3902/Sprintfinity3902.csproj:17]: When I tried to make a Hud folder initially it yelled at me that I couldn't use that name? Perhaps it has something to do with this... but I am honestly not sure how it happened.

File: Sprintfinity3902/PauseMenu.cs
Comment (Bailey-Ardrey) [reply Sprintfinity3902/PauseMenu.cs:17]: The "Map" here is the Sprint 3 picture of the map that we used to teleport to each room. It is no longer needed for Sprint 4!

**Comment by Bailey-Ardrey** [comment]
In my final two commits, I take care of Magic numbers in the InGameHud and related classes. I also removed the now redundant "AddIcons()" method in favor of simply "Initialize()".

**Review by kylegessel — APPROVED** [review]
Changes look great! Merging.

**Pull request merged**
